sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/ServerBatchRequest.java

```diff
@@ -59,27 +59,31 @@
         final ArrayNode arrayNode =  Utils.getSimpleObjectMapper().createArrayNode();
 
         for(CosmosItemOperation operation : operations) {
+            JsonSerializable operationJsonSerializable;
+
             if (operation instanceof ItemBatchOperation<?>) {
-                final ItemBatchOperation<?> itemBatchOperation = (ItemBatchOperation<?>) operation;
-                final JsonSerializable operationJsonSerializable = itemBatchOperation.serializeOperation();
+                operationJsonSerializable = ((ItemBatchOperation<?>) operation).serializeOperation();
+            } else if (operation instanceof ItemBulkOperation<?>) {
+                operationJsonSerializable = ((ItemBulkOperation<?>) operation).serializeOperation();
+            } else {
+                throw new UnsupportedOperationException("Unknown CosmosItemOperation.");
+            }
 
-                // TODO(rakkuma): If the string contains unicode the byte encoding len will be more. Fix it.
-                // Issue: https://github.com/Azure/azure-sdk-for-java/issues/16112
-                final int operationSerializedLength = operationJsonSerializable.toString().length();
+            // TODO(rakkuma): If the string contains unicode the byte encoding len will be more. Fix it.
```

Comment:
Please fix this before check-in

Reply:
At most of the places I can see in try/catch we put Exception instead of CosmosException. May be if their are some argument issue, or some validation stuff which didn't go to backend?

Reply:
Done. Have added test cases too to verify size overflow. Thanks.